dhall/src/Dhall/TypeCheck.hs

```diff
@@ -802,18 +802,20 @@ typeWithA tpa = loop
 
         case _R of
             Record ktsR -> do
-                _ <- fmap Dhall.Core.normalize (loop ctx t)
+                _ <- loop ctx t
 
                 case Dhall.Core.normalize t of
-                    Record ktsT -> do
-                        let keysR = Dhall.Set.fromList (Dhall.Map.keys ktsR)
-                        let keysT = Dhall.Set.fromList (Dhall.Map.keys ktsT)
-
-                        case Dhall.Set.difference keysT keysR of
-                            k : _ -> do
-                                Left (TypeError ctx e (MissingField k t))
-                            [] -> do
-                                return (Record ktsT)
+                    nf_t@(Record ktsT) -> do
+                        let projection = Record (Dhall.Map.intersection ktsR ktsT)
+
+                        t' <- loop ctx projection
+                        if Dhall.Core.judgmentallyEqual t t'
```

Comment:
Did you mean for this to be `Dhall.Core.judgmentallyEqual t projection`?

For example, for the expression:

```dhall
{ foo = 1 }.{ foo : Natural }
```

.... `t` would be `{ foo : Natural }`, but `t'` would be `Type`

Reply:
Thanks!

Reply:
You're welcome! 🙂